### PR TITLE
show coords std names

### PR DESCRIPTION
**Pull request by malmans2**
Tiny change.
In this case:
```python
from cf_xarray.tests.datasets import airds
airds.rename_dims(lon="x", lat="y").reset_coords().cf.describe()
```
currently `longitude` and `latitude` are not shown by `.describe()`.
This change shows them in the Standard Names section.

### CODE REVIEW

**Comment by malmans2** [comment]
Or maybe `longitude`, `latitude`, `depth`, and `time` should always go in `ds.cf.coordinates` even if they are not in `ds.coords`?

**Comment by dcherian** [comment]
Add a * if they are in coords?

**Comment by malmans2** [comment]
Good idea!
Maybe * for `cf_xarray` should represent any "CF variable"?
So under Standard Names we show everything in `cf.standard_names`, but we add a star to any variable that also appears in Axes, Coordinates, or Cell Measures.

**Comment by malmans2** [comment]
```python
from cf_xarray.tests.datasets import popds
popds["TLONG"].attrs["standard_name"] = "foo"
popds["TLAT"].attrs["standard_name"] = "latitude"
```
```
Axes:
	X: ['nlon']
	Y: ['nlat']
	Z: []
	T: []

Coordinates:
	longitude: ['TLONG', 'ULONG']
	latitude: ['TLAT', 'ULAT']
	vertical: []
	time: []

Cell Measures:
	area: []
	volume: []

Standard Names:
	foo: [*'TLONG']
	latitude: [*'TLAT']
	sea_water_potential_temperature: ['TEMP']
	sea_water_x_velocity: ['UVEL']
```

**Comment by malmans2** [comment]
Actually, thinking a bit more about it https://github.com/xarray-contrib/cf-xarray/pull/150#issuecomment-770442581 doesn't look good.
Probably better something like:
```
Axes:
      * X: ['nlon']
      * Y: ['nlat']
      * Z: []
      * T: []

Coordinates:
      * longitude: ['TLONG', 'ULONG']
      * latitude: ['TLAT', 'ULAT']
      * vertical: []
      * time: []

Cell Measures:
      * area: []
      * volume: []

Standard Names:
	foo: ['TLONG']
	latitude: ['TLAT']
	sea_water_potential_temperature: ['TEMP']
	sea_water_x_velocity: ['UVEL']

```

**Comment by dcherian** [comment]
I don't think the latest one adds much info...

**Comment by malmans2** [comment]
Maybe then:
```
Standard Names:
	foo: ['TLONG']
      * latitude: ['TLAT']
	sea_water_potential_temperature: ['TEMP']
	sea_water_x_velocity: ['UVEL']
```
What about cases where's there is both?
Does this get too confusing?
```
Standard Names:
	foo: ['TLONG']
      * latitude: ['TLAT']
        latitude: ['another_lat']
	sea_water_potential_temperature: ['TEMP']
	sea_water_x_velocity: ['UVEL']
```

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -963,11 +965,16 @@ def describe(self):
...
+            else:
+                text += f"\t{key}: {measures[key] if key in measures else []}\n"
 
         text += "\nStandard Names:\n"
         for key, value in sorted(self.standard_names.items()):
-            if key not in coords:
+            if key in {**axes, **coords, **measures}:
+                text += f"{star}{key}: {value}\n"
+            else:
                 text += f"\t{key}: {value}\n"
```
Comment (malmans2) [inline cf_xarray/accessor.py]: Mmm maybe this is the most useful:
Show all keys and values in `axes`, `coordinates`, `cell_measures`, and `standard_names`.
Add a star to `cell_measures` and `standard_names` keys if they are also keys of `axes` and `coordinates` (and `cell_measures` for `standard_names`).

**Comment by dcherian** [comment]
How about avoiding the star row in `standard_names` so it still prints out `latitude` under "standard names" when there is a data variable with `"standard_name: latitude"`

**Comment by malmans2** [comment]
I think I'm not following... Do you mean to
1. only print `longitude` if it's not in `cf.coordinates` (So no star case)
2. always print `longitude` with not star, no matter if it's also shown in coordinates
3. show with no star `set(ds.cf.standard_names["longitude"]) - set(ds.cf.coordinates["longitude"])`

But what if `cf.standard_names["longitude"] != cf.coordinates["longitude"]`?
I think in case 1 `describe()` does not show all variables, case 2 we can just keep `describe` with no stars at all (or the stars should go to keys with xarray coords, so all `cf.axes` and `cf.coordinates` and eventually some cell measures?), case 3 seems confusing if `describe()` shows under Standard Names something different than `cf.standard_names`

**Comment by dcherian** [comment]
```
>>> popds.reset_coords("TLAT").cf.describe()

Coordinates:
      * X: ['nlon']  #  * marks dimension coordinates like xarray
      * Y: ['nlat']
        Z: []
        T: []

      longitude: ['TLONG', 'ULONG']
      latitude: ['ULAT']  # note no TLAT
      vertical: []
      time: []

      area: []
      volume: []

    # now a list of standard names
      foo: ['TLONG']  # TLONG is also here

Data Variables:
	latitude: ['TLAT']  # TLAT is here
	sea_water_potential_temperature: ['TEMP']
	sea_water_x_velocity: ['UVEL']
```

**Comment by malmans2** [comment]
Ahhhh yes!

Use also subtitles, but maybe CF Axes, CF Coordinates, Cell Measures, Standard Names?
Also I think we can have measures in data variables, so split data variables in Cell measures and Standard Names?

Maybe good to add bounds?

Happy to update `describe` tomorrow if you didn't code it yet.
Also, can `cf.__repr__` trigger `describe()`?

**Comment by dcherian** [comment]
I haven't implemented it. I like your suggestions,  including adding bounds. let's just try to limit vertical space (should we keep the empty lines `[]`?)

```
>>> popds.reset_coords("TLAT").cf.describe()

Coordinates:
  Axes: * X: ['nlon']  #  * marks dimension coordinates like xarray
        * Y: ['nlat']
         Z: []
         T: []

Coordinates:   longitude: ['TLONG', 'ULONG']
                 latitude: ['ULAT']  # note no TLAT
                 vertical: []
                 time: []

      area: []
      volume: []

    # now a list of standard names
      foo: ['TLONG']  # TLONG is also here

Data Variables:
	latitude: ['TLAT']  # TLAT is here
	sea_water_potential_temperature: ['TEMP']
	sea_water_x_velocity: ['UVEL']

```

> Also, can cf.__repr__ trigger describe()?

I couldn't get it to work. File an xarray issue?

**Comment by malmans2** [comment]
How about this, so we show all coords/axes/measures but we keep it more compact?
```
Coordinates:
  - CF Axes:
      * X: ['nlon']
      * Y: ['nlat']
        n/a: T, Z
  - CF Coordinates: 
        longitude: ['TLONG', 'ULONG']
        latitude:  ['ULAT']
        n/a: time, vertical
  - Cell Measures:
        n/a: area, volume
  - Standard Names:
        foo: ['TLONG']

Data Variables:
  - Cell Measures:
        n/a: area, volume
  - Standard Names:
        latitude: ['TLAT']
        sea_water_potential_temperature: ['TEMP']
        sea_water_x_velocity: ['UVEL']
```

**Comment by dcherian** [comment]
That looks nice! though let's put `n/a` on the RHS so `area, volume: n/a`

**Comment by malmans2** [comment]
Deal!

**Comment by malmans2** [comment]
Anything wrong with this?
```python
class CFAccessor:
    ...
    def describe(self):
        """
        Print a string repr to screen.
        """

        print(self.__repr__())

    def __repr__(self):
        ....

        return text
```
`cf.__repr__` looks good to me:
Screenshot from 2021-02-02 17-48-53

**Comment by dcherian** [comment]
wow i'm sure I tried that and it didn't work! great let's deprecate `describe` then

**Comment by malmans2** [comment]
```python
from cf_xarray.tests.datasets import airds, popds
print(airds.cf)
```
```
Coordinates:
  - CF Axes:
      * X: ['lon']
      * Y: ['lat']
      * T: ['time']
        Z: n/a

  - CF Coordinates:
      * longitude: ['lon']
      * latitude: ['lat']
      * time: ['time']
        vertical: n/a

  - Cell Measures:
        area: ['cell_area']
        volume: n/a

  - Standard Names:
      * latitude: ['lat']
      * longitude: ['lon']
      * time: ['time']

Data Variables:
  - Cell Measures:
        area, volume: n/a

  - Standard Names:
        air_temperature: ['air']

```
```python
print(popds.cf)
```
```
Coordinates:
  - CF Axes:
      * X: ['nlon']
      * Y: ['nlat']
        T, Z: n/a

  - CF Coordinates:
        longitude: ['TLONG', 'ULONG']
        latitude: ['TLAT', 'ULAT']
        time, vertical: n/a

  - Cell Measures:
        area, volume: n/a

  - Standard Names:
        n/a

Data Variables:
  - Cell Measures:
        area, volume: n/a

  - Standard Names:
        sea_water_x_velocity: ['UVEL']
        sea_water_potential_temperature: ['TEMP']
```

**Comment by malmans2** [comment]
* are added to all keys with dimensions only. Maybe better to add * only to CF Axes?

**Comment by dcherian** [comment]
Can we compress this a little for e.g.

```
- Cell Measures: area: ['cell_area']
                 volume: n/a
```

I think the extra `*` are OK

**Comment by malmans2** [comment]
How about this?
```
Coordinates:
  - CF Axes: * X: ['nlon']
             * Y: ['nlat']
               T, Z: n/a

  - CF Coordinates:   longitude: ['TLONG', 'ULONG']
                      latitude: ['TLAT', 'ULAT']
                      time, vertical: n/a

  - Cell Measures:   area, volume: n/a

  - Standard Names:   n/a

Data Variables:
  - Cell Measures:   area, volume: n/a

  - Standard Names:   sea_water_potential_temperature: ['TEMP']
                      sea_water_x_velocity: ['UVEL']

```

**Comment by malmans2** [comment]
Should we add anything on top?
E.g.,  `<xarray.Dataset/DataArray.cf>`...

**Comment by dcherian** [comment]
I think we can avoid adding a top line. Otherwise this looks great!

File: cf_xarray/accessor.py [hunk 2]
```diff
@@ -950,27 +950,67 @@ def describe(self):
...
-        text += "\nStandard Names:\n"
-        for key, value in sorted(self.standard_names.items()):
-            if key not in _COORD_NAMES:
-                text += f"\t{key}: {value}\n"
-
-        print(text)
+
+        warnings.warn(
+            "'obj.cf.describe()' will be removed in a future version. "
+            "Use instead 'print(obj.cf)''",
```
Comment (dcherian) [inline cf_xarray/accessor.py:956]:
```suggestion
            "Use instead 'repr(obj.cf)' or 'obj.cf' in a Jupyter environment.",
```

File: cf_xarray/tests/test_accessor.py [hunk 3]
```diff
@@ -26,17 +28,99 @@ def assert_dicts_identical(dict1, dict2):
...
+      - Cell Measures:
+            area, volume: n/a
+
+      - Standard Names:
+            air_temperature: ['air']
+    """
+    assert actual == dedent(expected)
+
+    # DataArray (Coordinates section same as Dataset)
+    assert airds.cf.__repr__().startswith(airds["air"].cf.__repr__())
```
Comment (dcherian) [inline cf_xarray/tests/test_accessor.py:68]: clever!

File: cf_xarray/tests/test_accessor.py [hunk 4]
```diff
@@ -26,17 +28,99 @@ def assert_dicts_identical(dict1, dict2):
...
-    )
-    assert actual == expected
+def test_repr():
+    # Dataset.
+    # Stars: axes, coords, and std names
+    actual = airds.cf.__repr__()
+    expected = """\
+    Coordinates:
+      - CF Axes:
+          * X: ['lon']
```
Comment (dcherian) [inline cf_xarray/tests/test_accessor.py:38]: i guess you have to push the change moving these one line up...

**Review by dcherian — APPROVED** [review]

**Pull request merged**